### PR TITLE
Port katex, buildTree, parseTree to @flow.

### PR DESCRIPTION
**Pull request by marcianx**

### CODE REVIEW

File: src/buildTree.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 import Settings from "./Settings";
 import Style from "./Style";
 
-const buildTree = function(tree, expression, settings) {
+import type ParseNode from "./ParseNode";
+import type domTree from "./domTree";
+
+const buildTree = function(
+    tree: ParseNode[],
+    expression: string | String,
```
Comment (kevinbarabash) [inline src/buildTree.js]: `string` should be enough.

File: katex.js [hunk 2]
```diff
@@ -14,11 +15,18 @@ import buildTree from "./src/buildTree";
...
+import type ParseNode from "./src/ParseNode";
+
 /**
  * Parse and build an expression, and place that expression in the DOM node
  * given.
  */
-let render = function(expression, baseNode, options) {
+let render = function(
+    expression: string,
+    baseNode: Node,
```
Comment (kevinbarabash) [inline katex.js]: Should this be `ParseNode`?

Comment (marcianx) [reply katex.js]: This is an HTML [`Node`](https://developer.mozilla.org/en-US/docs/Web/API/Node). Should it perhaps be an [`HTMLElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement) instead?

File: src/buildTree.js (hunk 1 above)
Comment (marcianx) [reply src/buildTree.js]: Done.

File: katex.js (hunk 2 above)
Comment (kevinbarabash) [reply katex.js]: I didn't read the surrounding code.  `Node` is fine since that's what `applyNode` is defined on.

**Pull request merged**